src/lib/features/integration-events/integration-events-service.ts

```diff
@@ -0,0 +1,57 @@
+import type { Logger } from '../../logger';
+import type { IFlagResolver, IUnleashConfig } from '../../types';
+import type {
+    IntegrationEventsStore,
+    IntegrationEventWriteModel,
+} from './integration-events-store';
+import type { IntegrationEventSchema } from '../../openapi/spec/integration-event-schema';
+
+export class IntegrationEventsService {
+    private readonly logger: Logger;
+    private integrationEventsStore: IntegrationEventsStore;
+    private flagResolver: IFlagResolver;
+
+    constructor(
+        {
+            integrationEventsStore,
+        }: { integrationEventsStore: IntegrationEventsStore },
+        {
+            getLogger,
+            flagResolver,
+        }: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>,
+    ) {
+        this.integrationEventsStore = integrationEventsStore;
+        this.flagResolver = flagResolver;
+        this.logger = getLogger('integration-events-service');
+    }
+
+    isEnabled(): boolean {
+        return this.flagResolver.isEnabled('integrationEvents');
+    }
+
+    async getPaginatedEvents(
+        id: number,
+        limit: number,
+        offset: number,
+    ): Promise<IntegrationEventSchema[]> {
+        return this.integrationEventsStore.getPaginatedEvents(
+            id,
+            limit,
+            offset,
+        );
+    }
+
+    async registerEvent(
```

Comment:
I think this function feels weird to me for a few reasons:

1) Because it needs to respect the feature flag, its return type can be undefined (glanced at the store, looks like the store returns an always defined type?). If the flag is the only reason we can return undefined I think that's incorrect
2) It's unclear at a glance why this needs to return the inserted object at all. Is that purely for tests?
3) Handling the flag at all here feels off to me, having functions that quietly do nothing if a flag is off feels incorrect to me unless they're at the outermost layers of the system (and even then I think I'd want logged warning or an error to be raised). This could make sense if this is the common layer where all integrations will hit and there's no other abstraction to handle this

Reply:
1. Yep, fixed in https://github.com/Unleash/unleash/pull/7614/commits/278f7c7fe2091607f130cc95386a5f167166ee69 - We'll handle this in `src/lib/addons/addon.ts` at a later stage.
2. Tests and because I think it's generally polite to return to newly created object, like we do for POST requests. It's especially nice when we have follow up actions, like doing something with its id. Might just be personal preference though.
3. Given that this will not be available through the controller layer, this would likely be the common outermost layer of this specific logic, however like we discussed and I mentioned above, we can pass down the `flagResolver` and handle this in the `Addon` base class.

> Handling the flag at all here feels off to me, having functions that quietly do nothing if a flag is off feels incorrect to me unless they're at the outermost layers of the system (and even then I think I'd want logged warning or an error to be raised)

Generally speaking, I'm not so sure I agree and I would like you to fight me here because I want to better understand your arguments. 

I prefer being able to call these new methods where they make sense and not having to worry about it. As the layer usually responsible for the core domain logic of a feature, the service should be smart about it and only actually do the thing if it's enabled. Otherwise, it is the caller's responsibility to ensure that this feature is always behind a feature flag. It's also more work to clean up the flag afterwards.

I also don't like the idea of logging warnings or raising errors because to me code that is flagged off should act the exact same (or as close as possible) as not existing at all: no-op.

However I 100% agree that it's icky to change method's signatures because of flags, which is why I like our new approach 🙂